winpr/include/winpr/ndr.h

```diff
@@ -0,0 +1,222 @@
+/**
+ * FreeRDP: A Remote Desktop Protocol Implementation
+ * Authentication redirection virtual channel
+ *
+ * Copyright 2024 David Fort <contact@hardening-consulting.com>
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+#ifndef WINPR_NDR_H_
+#define WINPR_NDR_H_
+
+#include <winpr/stream.h>
+
+#define NDR_PTR_NULL (0UL)
+
+#define NDR_SIMPLE_TYPE_DECL(LOWER, UPPER)                                                  \
+	WINPR_API BOOL ndr_read_##LOWER(NdrContext* context, wStream* s, UPPER* v);             \
+	WINPR_API BOOL ndr_read_##LOWER##_(NdrContext* context, wStream* s, const void* hints,  \
+	                                   void* v);                                            \
+	WINPR_API BOOL ndr_write_##LOWER(NdrContext* context, wStream* s, UPPER v);             \
+	WINPR_API BOOL ndr_write_##LOWER##_(NdrContext* context, wStream* s, const void* hints, \
+	                                    const void* v);                                     \
+	extern WINPR_API const NdrMessageDescr ndr_##LOWER##_descr_s;                           \
+	WINPR_API NdrMessageType ndr_##LOWER##_descr(void)
+
+#define NDR_ARRAY_OF_TYPE_DECL(TYPE, UPPERTYPE)                                                  \
```

Comment:
maybe also remove the macros from public headers.

Reply:
The macros are used in this header. And usually when you want to declare a new type you'll want to use them.